homeassistant/data_entry_flow.py

```diff
@@ -26,6 +26,8 @@
 # Event that is fired when a flow is progressed via external or progress source.
 EVENT_DATA_ENTRY_FLOW_PROGRESSED = "data_entry_flow_progressed"
 
+FlowInput = Mapping[str, Any]
```

Comment:
I hesitated about making it optional `FlowInput = Optional[Mapping[str, Any]]` but I think some steps (like `async_step_reauth`) have it compulsory. We could also create a separate type: `OptionalFlowInput = Optional[FlowInput]`

Reply:
I think `user_input: FlowInput | None = None` is very clear